pkg/raft/raft.go

```diff
@@ -1997,9 +2007,12 @@ func (r *raft) handleHeartbeat(m pb.Message) {
 	// commit index if accTerm >= m.Term.
 	// TODO(pav-kv): move this logic to raftLog.commitTo, once the accTerm has
 	// migrated to raftLog/unstable.
-	mark := LogMark{Term: m.Term, Index: min(m.Commit, r.raftLog.lastIndex())}
-	if mark.Term == r.raftLog.accTerm() {
-		r.raftLog.commitTo(mark)
+	// TODO(iskettaneh): Remove this logic once no longer needed.
+	if m.Commit != 0 {
+		mark := LogMark{Term: m.Term, Index: min(m.Commit, r.raftLog.lastIndex())}
+		if mark.Term == r.raftLog.accTerm() {
+			r.raftLog.commitTo(mark)
+		}
 	}
```

Comment:
Probably don't need to change this bit and the TODO, because `commitTo` is a no-op if `Commit` <= the current commit index, which is trivially true when `Commit == 0`.